Fixed #11593 -- Added testing to resuable apps intro (#18018)
django/django: docs/intro/reusable-apps.txt

#. Run the development server to confirm the project continues to work.

Testing your app
================

> Since we moved the ``polls`` directory out of the project,

Reviewer: Blank line is missing:
```suggestion

Since we moved the ``polls`` directory out of the project,
```

we don't have the project's ``settings.py`` and ``urls.py`` anymore.
We need another way to configure our app in order to test it.

> #. First we create a ``tests`` directory and move the tests over from the

Reviewer: `we` sounds unnecessary in all points.

> Since we moved the ``polls`` directory out of the project,
> we don't have the project's ``settings.py`` and ``urls.py`` anymore.
> We need another way to configure our app in order to test it.
> 
> #. First create a ``tests`` directory and move the tests over from the
>    ``polls`` directory. Because the tests now reside in a different module
>    the relative import ``from .models import Poll`` has to be changed to
>    ``from polls.models import Poll``.

Reviewer 2: Also note @RaphaelKimmig that all lines should be wrapped at 79cols, so please check this and all the following changes to match this style:
```suggestion
Since we moved the ``polls`` directory out of the project, we don't have the
project's ``settings.py`` and ``urls.py`` anymore. We need another way to
configure our app in order to test it.

#. First create a ``tests`` directory and move the tests over from the
   ``polls`` directory. Because the tests now reside in a different module the
   relative import ``from .models import Poll`` has to be changed to
   ``from polls.models import Poll``.
```

Author: Alright, reformatted :-)

> #. Now we create a ``settings.py`` inside the ``tests`` folder which will contain

Reviewer: Wrap docs at 79 chars.

#. Now create a ``settings.py`` inside the ``tests`` folder which will contain
   only the settings required to test the application.

   .. code-block:: python
       :caption: ``tests/settings.py``

       from django.urls import include, path

       SECRET_KEY = "fake-key"

       INSTALLED_APPS = [
           "polls",

>            # adding tests allows us to define models inside the tests package
>            # "tests",

Reviewer 3: ```suggestion
           "tests",  # Optional: adding tests allows us to define models inside the tests directory.
```
(maybe) note that this file is also in the advanced topic so if we chose to have this in both places, this comment applies there too

       DATABASES = {
           "default": {
               "ENGINE": "django.db.backends.sqlite3",
               "NAME": ":memory:",
           }
       }

       TEMPLATES = [
           {
               "BACKEND": "django.template.backends.django.DjangoTemplates",
               "APP_DIRS": True,
           },
       ]

       ROOT_URLCONF = "tests.urls"

#. Because we test our views, we need a :setting:`ROOT_URLCONF` that includes
   our app. We create an appropriate ``urls.py`` inside the ``tests``
   directory.

   .. code-block:: python
       :caption: ``tests/urls.py``

       from django.urls import include, path

       urlpatterns = [
           path("polls/", include("polls.urls")),
       ]

#. Now we can run the tests.

   .. code-block:: shell

        python -m django test --settings=tests.settings tests

For more information on testing reusable applications see :ref:`Advanced testing topics
<testing-reusable-applications>`.

> 

Reviewer: ```suggestion
```
